packages/eas-cli/src/commands/update/index.ts

```diff
@@ -586,15 +608,32 @@ async function getRuntimeVersionObjectAsync(
     }
   }
 
-  return Object.fromEntries(
-    platforms.map(platform => [
-      platform,
-      nullthrows(
-        Updates.getRuntimeVersion(exp, platform),
-        `Unable to determine runtime version for ${platform}`
-      ),
-    ])
-  );
+  try {
+    return transformRuntimeVersions(exp, platforms);
+  } catch (error: any) {
+    ora().start().fail(error.message);
+
+    const runConfig = await selectAsync(`Do you want us to run 'eas update:configure' for you?`, [
+      { title: 'Yes', value: true },
+      {
+        title: 'No, I will edit files or run the command manually (EAS CLI exits)',
+        value: false,
+      },
+    ]);
+
+    if (!runConfig) {
+      Errors.exit(1);
+    }
+
+    const { default: UpdateConfigure } = await import('./configure');
```

Comment:
this is the first time we've done this in the CLI so it may be setting a bit of a precedent. I see that https://oclif.io/docs/running_programmatically says it's doable, but it also mentions that factoring it out is preferred and I tend to agree. The main reason I think we should factor out the underlying code is that that there's no typechecking on this passing of flags, so if we add a flag to the configure command, we'll most likely forget to inspect this callsite to ensure the flag is passed. By factoring out the code, we can type the method strictly.

Reply:
According to dynamic imports, Expo CLI use this pattern for the performance, I also saw one usage in this codebase already for loading config file:
* https://github.com/expo/eas-cli/blob/024c699637f46914d0ba9766f321faaf5fc4efb4/packages/eas-cli/src/metadata/config.ts#L23

Reply:
Ah, sorry, I didn't mean the dynamic import, I meant the calling of another command programmatically instead of just factoring out the code. Dynamic imports are totally cool.